Mathlib/NumberTheory/ArithmeticFunction.lean

```diff
@@ -1422,6 +1424,91 @@ theorem prod_eq_iff_prod_pow_moebius_eq_on_of_nonzero [CommGroupWithZero R]
 
 end SpecialFunctions
 
+section Sum
+
+theorem sum_zeta (N : ℕ) : ∑ n ∈ Icc 0 N, zeta n = N := by
+  simp only [zeta_apply, sum_ite, sum_const_zero, sum_const, smul_eq_mul, mul_one, zero_add]
+  rw [show {x ∈ Icc 0 N | ¬x = 0} = Ioc 0 N by ext; simp; cutsat]
+  simp
+
+/-- Useful lemma for reordering sums. -/
+lemma divisorsAntidiagonal_of_le_eq_prod_filter {n N : ℕ} (n_ne_zero : n ≠ 0) (hn : n ≤ N) :
```

Comment:
This lemma is not about arithmetic functions. It should go in `Mathlib.NumberTheory.Divisors`